tests/Functional/Execution/AbstractTest.php

```diff
@@ -71,6 +200,327 @@ public function testResolveFunctionForInterface()
 
         $this->assertEquals($expected->getData(), $result->getData());
     }
+
+    /**
+     * resolveType on Interface yields useful error
+     *
+     * @throws \Digia\GraphQL\Error\InvariantException
+     * @throws \Digia\GraphQL\Error\SyntaxErrorException
+     */
+    public function testResolveTypeOnInterfaceYieldsUsefulError()
+    {
+        $PetInterfaceType = GraphQLInterfaceType([
+            'name'        => 'Pet',
+            'resolveType' => function ($result, $context, $info) use (&$DogType, &$CatType, &$HumanType) {
+                if ($result instanceof Dog) {
+                    return $DogType;
```

Comment:
The Apollo implementation supports just returning the name of the type. This would probably be a better approach to ensure that you don't have multiple instances of the same types.

Reply:
Please check the test below `resolveType allows resolving with type name`